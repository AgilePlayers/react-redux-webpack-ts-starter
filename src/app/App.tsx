import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { hot } from 'react-hot-loader';
import HeaderComponent from './common/header/HeaderComponent';
import FooterComponent from './common/footer/FooterComponent';
import HomeComponent from './home/HomeComponent';
import BookListContainer from './books/BookListContainer';

import './common/common.scss';

interface AppProps {
  store: Store;
}

const App: React.SFC<AppProps> = ({ store }) => (
  <Provider store={store}>
    <Router>
      <>
        <div className="container">
          <HeaderComponent />
          <div className="content">
            <Route exact path="/" component={HomeComponent} />
            <Route exact path="/books" component={BookListContainer} />
          </div>
        </div>
        <FooterComponent />
      </>
    </Router>
  </Provider>
);

export default hot(module)(App);
// export default App;
