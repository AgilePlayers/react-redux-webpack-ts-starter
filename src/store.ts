import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducers from './reducers';

const logger = createLogger();
const middleware = applyMiddleware(thunk, logger);
const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const enhancers = composeEnhancers(middleware);

const store = createStore(reducers, enhancers);

if (process.env.NODE_ENV === 'development' && module.hot) {
  module.hot.accept('./reducers', async () => {
    // const nextRootReducer = require('./reducers.js').default;
    const nextRootReducer = (await import('./reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export default store;
