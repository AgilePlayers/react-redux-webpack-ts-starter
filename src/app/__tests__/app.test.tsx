import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import App from '../App';
import store from '../../store';

jest.mock('../../store', () => ({
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(),
}));

describe('app / App', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<App store={store} />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
