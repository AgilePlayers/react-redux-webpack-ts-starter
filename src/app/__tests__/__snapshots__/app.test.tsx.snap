// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app / App should match snapshot 1`] = `ShallowWrapper {}`;
